main.py

```diff
@@ -80,28 +89,55 @@ def __val_freq(val):
         return {"freq": int(val) * 1000000}
 
 
+def tegrastats_worker(interval):
+    """Thread worker to run `tegrastats` and populate queue
+
+    Args:
+        interval: data collection interval in seconds
+    """
+    interval = interval * 1000
+    logging.info(f"starting tegrastats worker [{interval} ms]...")
+
+    try:
+        with subprocess.Popen(
```

Comment:
I noticed you run the threads with daemon=False and you use a stop event. Generally, I think this is the right thing to do. The only part I'm not sure about is, in principle the .readline _could_ block forever if something goes wrong.

I'm not sure if you would prefer to change daemon=True which allows to program to exit immediately, even if there are background threads. (You may have noticed that Python just sits there if there around background threads running by default.)

Again, I'd only recommend that in a case like this where you basically have a background worker that runs for the entire program duration... otherwise knowing how to stop and clean up is important.

Reply:
Testing as a `deamon=True` and killing the process with a Ctrl+C from within the container and killing container itself cleaned up any running subprocess calls (tegrastats, jetson_clocks).  So, yeah I did this change.